src/System.Net.Requests/tests/RequestStreamTest.cs

```diff
@@ -107,6 +133,77 @@ public void WriteAsync_TokenIsCanceled_TaskIsCanceled()
             }
         }
 
+        #endregion
+
+        #region BeginWrite
+
+        private IAsyncResult BeginWriteWrapper(Stream stream, byte[] buffer, int offset, int count)
+        {
+            AsyncCallback callback = null;
+            object state = new object();
+            IAsyncResult result = stream.BeginWrite(buffer, offset, count, callback, state);
+            stream.EndWrite(result);
+
+            return result;
+        }
+
+        [Fact]
+        public void BeginWriteAsync_BufferIsNull_ThrowsArgumentNullException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentNullException>(() => BeginWriteWrapper(stream, null, 0, 1));
```

Comment:
Changing everywhere... I continued following the original code that  didn't check for the parameter name... thanks for suggesting, will use this going ahead for any sort of Assert.Throws wherever possible. And then yes, BeginWriteWrapper doubly doesn't make sense.